pydantic/main.py

```diff
@@ -360,11 +360,15 @@ def __init__(__pydantic_self__, **data: Any) -> None:
         values, fields_set, validation_error = validate_model(__pydantic_self__.__class__, data)
         if validation_error:
             raise validation_error
+        if values is None:
```

Comment:
Good idea! 👍 
But what if you do `return 3` in your `root_validator` you will still have an implicit error `__dict__ must be set to a dictionary, not a 'int'`! Maybe we could handle a more generic case with something like
```python
try:
    object_setattr(__pydantic_self__, '__dict__', values)
except TypeError as e:
    raise TypeError('Did you return a valid dict in your root validators ?') from e
```
?
Just my 2 cents :)

Reply:
Thanks for the feedback!  This looks good -- updated the PR 👍